features/GroupInvites/GroupInvite.screen.tsx

```diff
@@ -1,81 +1,88 @@
+import { navigate } from "@utils/navigation";
 import { GroupWithCodecsType } from "@utils/xmtpRN/client";
+import { refreshGroup } from "@utils/xmtpRN/conversations";
 import { useCallback, useEffect, useRef, useState } from "react";
 import { StyleSheet, Text, View, useColorScheme } from "react-native";
 import { ActivityIndicator } from "react-native-paper";
 import { useSafeAreaInsets } from "react-native-safe-area-context";
 
-import { NavigationParamList } from "./Navigation/Navigation";
-import Button from "../components/Button/Button";
-import { useCurrentAccount } from "../data/store/accountsStore";
-import { navigate } from "../utils/navigation";
-import { refreshGroup } from "../utils/xmtpRN/conversations";
+import {
+  getInviteJoinRequest,
+  saveInviteJoinRequest,
+} from "./groupInvites.utils";
 
 export default function GroupInviteScreen({
   route,
   navigation,
 }: NativeStackScreenProps<NavigationParamList, "GroupInvite">) {
+  /**************************************************************
+   * View/UI
+   * ************************************************************/
   const styles = useStyles();
+  const colorScheme = useColorScheme();
+  const insets = useSafeAreaInsets();
+  const headerHeight = useHeaderHeight();
+
+  /**************************************************************
+   * Zustand
+   * ************************************************************/
   const account = useCurrentAccount() as string;
+
+  /**************************************************************
+   * React Query
+   * ************************************************************/
   const {
     data: groupInvite,
     error: groupInviteError,
     isLoading,
   } = useGroupInviteQuery(route.params.groupInviteId);
-  const [
-    { polling, finishedPollingUnsuccessfully, joinStatus },
-    setGroupJoinState,
-  ] = useState<{
-    polling: boolean;
-    /* Unsuccessfully here means that we have not yet received confirmation 
-        from the invitee that we have been added to the group */
-    finishedPollingUnsuccessfully: boolean;
-    joinStatus: null | GroupJoinRequestStatus;
-  }>({
-    polling: false,
-    finishedPollingUnsuccessfully: false,
-    joinStatus: null,
-  });
-  const colorScheme = useColorScheme();
+
+  /**************************************************************
+   * State Machine
+   * ************************************************************/
+  const [state, send] = useActor(joinGroupService);
+
   const [newGroup, setNewGroup] = useState<GroupWithCodecsType | undefined>(
     undefined
   );
+
   const { allowGroup } = useGroupConsent(newGroup?.topic || "");
   const handlingNewGroup = useRef(false);
 
   const handleNewGroup = useCallback(
     async (group: GroupWithCodecsType) => {
-      if (group.client.address === account && !handlingNewGroup.current) {
+      const whatDoesThisConditionIndicate =
+        group.client.address === account && !handlingNewGroup.current;
```

Comment:
This is checking if the new group is the correct account of what's expected from the user
There's also a check to make sure it's not doing multiple groups
I think both are a bit hacky and can probably be removed

Reply:
Thanks for answering all my questions.

Two follow ups here: 

1) what does `group.client` mean? 

2) When could `client.address` NOT equal the current account?

Reply:
There is a client for each connected account, but I don't think that's really a case in this code, so I think it should be removed